### PR TITLE
Megamek

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/MegaMek/megamek/issues/4154
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build org.megamek.MegaMek

**Comment by Mailaender** [comment]
bot, build org.megamek.MegaMek

**Comment by hfiguiere** [comment]
bot, build org.megamek.MegaMek

File: org.megamek.MegaMek.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+app-id: org.megamek.MegaMek
+runtime: org.freedesktop.Platform
+runtime-version: 22.08
+sdk: org.freedesktop.Sdk
+command: megamek
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk11
+finish-args:
+  - --socket=wayland
```
Comment (hfiguiere) [inline org.megamek.MegaMek.yaml:9]: does it really work with wayland?

My experience with Java is that usually it doesn't.

File: org.megamek.MegaMek.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        sha256: a8e88a19133be925b3763d7a056e6ed9335e8c4f90232baa3c75b00090ae5532
+      - type: file
+        path: megamek.desktop
+      - type: file
+        path: org.megamek.MegaMek.metainfo.xml
+      - type: script
+        dest-filename: megamek.sh
+        commands:
+          - cd /app/share/megamek/
+          - /app/jre/bin/java -jar MegaMek.jar
```
Comment (hfiguiere) [inline org.megamek.MegaMek.yaml:50]:
```suggestion
          - exec /app/jre/bin/java -jar MegaMek.jar
```

File: org.megamek.MegaMek.yaml (hunk 1 above)
Comment (Mailaender) [reply org.megamek.MegaMek.yaml:9]: I haven't tried. It uses the Swing UI toolkit.

**Comment by Mailaender** [comment]
bot, build org.megamek.MegaMek

**Pull request closed without merging**